v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java

```diff
@@ -0,0 +1,218 @@
+                        .build(),
+                    uploadAvroFile(
+                        jobInfo,
+                        TABLE1,
+                        "cdc1.avro",
+                        "DataStreamToSpannerSimpleIT/mysql-cdc-Users.avro"),
+                    SpannerRowsCheck.builder(spannerResourceManager, TABLE1)
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build()))
+            .build();
+
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(8)), conditionCheck);
+
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+
+    // Assert specific rows
+    assertSpannerRowContent1();
+  }
+
+  private void assertSpannerRowContent1() {
+    List<Map<String, Object>> events = new ArrayList<>();
+
+    Map<String, Object> row1 = new HashMap<>();
+    row1.put("id", 1);
+    row1.put("name", "Tester Kumar");
+    row1.put("age_spanner", 50);
+
+    Map<String, Object> row2 = new HashMap<>();
+    row2.put("id", 3);
+    row2.put("name", "Tester Gupta");
+    row2.put("age_spanner", 25);
+
+    Map<String, Object> row3 = new HashMap<>();
+    row3.put("id", 4);
+    row3.put("name", "Tester");
+    row3.put("age_spanner", 30);
+    events.add(row1);
+    events.add(row2);
+    events.add(row3);
+
+    SpannerAsserts.assertThatStructs(
+            spannerResourceManager.runQuery("select * from Users where id in (1, 3, 4)"))
+        .hasRecordsUnorderedCaseInsensitiveColumns(events);
+  }
+
+  @Test
+  public void migrationTestWithInsertsOnly() {
+    // Construct a ChainedConditionCheck with 4 stages.
+    // 1. Send initial wave of events
+    // 2. Wait on Spanner to have events
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    uploadAvroFile(
+                        jobInfo,
+                        TABLE2,
+                        "backfill.avro",
+                        "DataStreamToSpannerSimpleIT/mysql-backfill-Movie.avro"),
+                    uploadAvroFile(
+                        jobInfo,
+                        TABLE2,
+                        "cdc1.avro",
+                        "DataStreamToSpannerSimpleIT/mysql-cdc-Movie.avro"),
+                    SpannerRowsCheck.builder(spannerResourceManager, TABLE2)
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build()))
+            .build();
+
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(8)), conditionCheck);
+
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+
+    // Assert specific rows
+    assertSpannerRowContent2();
+  }
+
+  private void assertSpannerRowContent2() {
```

Comment:
Nit: `Content1` / `Content2` is not readable - please rename appropriately.

Reply:
Done.